pipcompilemulti/features/index.py

```diff
@@ -0,0 +1,30 @@
+"""
+Enable indexes
+==============
+
+This flag mimics the logic of the ``pip-compile`` ``--index`` and ``--no-index`
+flag by opting to add or not add the ``pip`` index to the generated files.
+
+.. code-block:: text
+
+    --index / --no-index     Add index URL to generated files (default false)
+
+Note the default behavior is not to add the index, i.e., ``--no-index`.
+"""
+
+from .base import ClickOption
+from .forward import ForwardOption
+
+
+class Index(ForwardOption):
+    """Optionally add the index URL to the generated files."""
+
+    OPTION_NAME = "index"
+    CLICK_OPTION = ClickOption(
+        long_option="--index/--no-index",
```

Comment:
I feel it's clearer to have `--annotate-index/--no-annotate-index` as was proposed in https://github.com/jazzband/pip-tools/issues/373.
pip-compile-multi doesn't have the constraint of backward compatibility that prevented pip-tools from clarifying the flag.
Let's also rename class to `AnnotateIndex`